plugins/catalog-backend-module-github/src/providers/GitHubEntityProvider.ts

```diff
@@ -0,0 +1,271 @@
+   * to enable automatic scheduling of tasks.
+   */
+  schedule: TaskRunner;
+
+  /**
+   * The logger to use.
+   */
+  logger: Logger;
+}
+
+/**
+ * Discovers catalog files located in [GitHub](https://github.com).
+ * The provider will search your GitHub account and register catalog files matching the configured path
+ * as Location entity and via following processing steps add all contained catalog entities.
+ * This can be useful as an alternative to static locations or manually adding things to the catalog.
+ *
+ * @public
+ */
+export class GitHubEntityProvider implements EntityProvider {
+  private readonly config: GitHubEntityProviderConfig;
+  private readonly logger: Logger;
+  private readonly integration: GitHubIntegrationConfig;
+  private readonly scheduleFn: () => Promise<void>;
+  private connection?: EntityProviderConnection;
+  private readonly githubCredentialsProvider: GithubCredentialsProvider;
+
+  static fromConfig(
+    config: Config,
+    options: GitHubEntityProviderOptions,
+  ): GitHubEntityProvider[] {
+    const integrations = ScmIntegrations.fromConfig(config);
+    const integration = integrations.github.byHost('github.com');
+
+    if (!integration) {
+      throw new Error(
+        `There is no GitHub config that matches github. Please add a configuration entry for it under integrations.github`,
+      );
+    }
+
+    return readProviderConfigs(config).map(
+      providerConfig =>
+        new GitHubEntityProvider(
+          providerConfig,
+          integration,
+          options.logger,
+          options.schedule,
+        ),
+    );
+  }
+
+  private constructor(
+    config: GitHubEntityProviderConfig,
+    integration: GitHubIntegration,
+    logger: Logger,
+    schedule: TaskRunner,
+  ) {
+    this.config = config;
+    this.integration = integration.config;
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(schedule);
+    this.githubCredentialsProvider =
+      SingleInstanceGithubCredentialsProvider.create(integration.config);
+    this.scheduleFn = this.createScheduleFn(schedule);
+  }
+
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.getProviderName} */
+  getProviderName(): string {
+    return `github-provider:${this.config.id}`;
+  }
+
+  /** {@inheritdoc @backstage/plugin-catalog-backend#EntityProvider.connect} */
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    return await this.scheduleFn();
+  }
+
+  private createScheduleFn(schedule: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return schedule.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+          try {
+            await this.refresh();
```

Comment:
you should pass the logger here. Currently, the new logger instance is unused.

Reply:
resolved